apps/portal/src/App.js

```diff
@@ -191,10 +192,18 @@ export default class App extends React.Component {
     /** Initialize portal setup on load, fetch data and setup state*/
     async initSetup() {
         try {
+            // i18n should be initialized in only one location, so create the instance before performing any async operations
+            const baseLocale = (this.props.siteI18nEnabled && this.props.locale) || 'en';
+            const i18n = createAsyncInstance('portal', baseLocale, this.props.localeRoot);
+
             // Fetch data from API, links, preview, dev sources
             const {site, member, page, showPopup, popupNotification, lastPage, pageQuery, pageData} = await this.fetchData();
-            const i18nLanguage = this.props.siteI18nEnabled ? this.props.locale || site.locale || 'en' : 'en';
-            const i18n = i18nLib(i18nLanguage, 'portal');
+
+            const localeWithSite = this.props.siteI18nEnabled && !this.props.locale && site.locale;
+
+            if (localeWithSite && baseLocale !== localeWithSite) {
+                i18n.changeLanguage(localeWithSite);
+            }
```

Comment:
If I've understood correctly, this will cause the site locale setting to override what might be passed in as a prop.  Is there a compelling reason for doing so?  I suggest that if a locale is passed in as a prop, it should dominate over the site-wide setting, which allows for loading portal in a language other than the sitewide locale for some fraction of the site.  (While that's not full multi-language support, it's a good start.)

Reply:
`localeWithSite` takes the data attribute into consideration🙂 

```javascript
const localeWithSite = this.props.siteI18nEnabled && !this.props.locale && site.locale;
//                                                   ^ Passing a locale will make result `false`
```

Reply:
Ah! Sorry, I missed that! :)